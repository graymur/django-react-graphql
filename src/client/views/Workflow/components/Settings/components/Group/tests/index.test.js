import React from 'react'
import renderer from 'react-test-renderer'
import { Group } from '../index'
import defaultProps from './data'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

test('Renders empty option and validation error, submit button disabled', () => {
	const component = renderer.create(<Group {...defaultProps} />)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})

test("Renders selected option, doesn't render validation option, submit button enabled", () => {
	const props = {
		...defaultProps,
		widgetsValues: { listId: 1 },
		widgetsErrors: { listId: false },
	}

	const component = renderer.create(<Group {...props} />)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})

test('handles onChange event', () => {
	const setWidgetValue = jest.fn()

	const component = mount(
		<Group {...defaultProps} setWidgetValue={setWidgetValue} />,
	)

	component
		.find('.workflow_select_mailing')
		.simulate('change', { target: { value: '' } })
	component
		.find('.workflow_select_mailing')
		.simulate('change', { target: { value: 2 } })
	expect(setWidgetValue.mock.calls.length).toBe(2)
	// result of call with empty value should have validation error
	expect(setWidgetValue.mock.calls[0]).toEqual([
		'listId',
		'',
		{ code: 1, messageWidgetId: 'list_id-error' },
	])
	// result of call with valid value should not have validation error
	expect(setWidgetValue.mock.calls[1]).toEqual(['listId', 2, false])
})

test('ignored onSave event on disabled "save" button', () => {
	const onSave = jest.fn()

	const component = mount(<Group {...defaultProps} onSave={onSave} />)
	component.find('._save-block .main-button').simulate('click')
	expect(onSave.mock.calls.length).toBe(0)
})

test('ignored onSave event on disabled "save" button', () => {
	const onSave = jest.fn()
	const props = {
		...defaultProps,
		widgetsValues: {
			...defaultProps.widgetsValues,
			listId: 1,
		},
		widgetsErrors: { listId: false },
	}

	const component = mount(<Group {...props} onSave={onSave} />)
	component.find('._save-block .main-button').simulate('click')
	expect(onSave.mock.calls.length).toBe(1)
	expect(onSave.mock.calls[0]).toEqual([
		{ appName: 'cn_mailchimp', values: { listId: 1 }, groupIndex: 0 },
	])
})
