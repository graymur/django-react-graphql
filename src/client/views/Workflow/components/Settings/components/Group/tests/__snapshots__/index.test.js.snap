// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders empty option and validation error, submit button disabled 1`] = `
<div
  className="workflow-settings__group"
>
  <h3
    className="workflow-settings__group__type"
  >
    Trigger
  </h3>
  <div
    className="workflow-settings__group__wr _active"
  >
    <header
      className="workflow-settings__group__header"
      onClick={[Function]}
    >
      <span
        className="app-logo app-logo-cn_mailchimp"
        title={undefined}
      >
        <img
          src="undefined/mailchimp_128x128.png"
        />
      </span>
      <h2
        className="workflow-settings__group__title"
      >
        Create or update contact
      </h2>
    </header>
    <div
      className="workflow-widget _select"
    >
      <select
        className="workflow-widget__select styled-select workflow_select_mailing"
        defaultValue={undefined}
        id="mailing_list_id"
        name="listId"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Select
        </option>
        <option
          value={1}
        >
          Option 1
        </option>
        <option
          value={2}
        >
          Option 2
        </option>
      </select>
    </div>
    <div
      className="workflow-widget _validation-message"
    >
      <div
        className="workflow-widget__validation-message _active"
      >
        Select mailing list
      </div>
    </div>
    <div
      className="workflow-widget _save-block"
    >
      <button
        className="main-button"
        disabled={true}
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
</div>
`;

exports[`Renders selected option, doesn't render validation option, submit button enabled 1`] = `
<div
  className="workflow-settings__group"
>
  <h3
    className="workflow-settings__group__type"
  >
    Trigger
  </h3>
  <div
    className="workflow-settings__group__wr _active"
  >
    <header
      className="workflow-settings__group__header"
      onClick={[Function]}
    >
      <span
        className="app-logo app-logo-cn_mailchimp"
        title={undefined}
      >
        <img
          src="undefined/mailchimp_128x128.png"
        />
      </span>
      <h2
        className="workflow-settings__group__title"
      >
        Create or update contact
      </h2>
    </header>
    <div
      className="workflow-widget _select"
    >
      <select
        className="workflow-widget__select styled-select workflow_select_mailing"
        defaultValue={1}
        id="mailing_list_id"
        name="listId"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Select
        </option>
        <option
          value={1}
        >
          Option 1
        </option>
        <option
          value={2}
        >
          Option 2
        </option>
      </select>
    </div>
    <div
      className="workflow-widget _validation-message"
    >
      <div
        className="workflow-widget__validation-message"
      >
        Select mailing list
      </div>
    </div>
    <div
      className="workflow-widget _save-block"
    >
      <button
        className="main-button"
        disabled={false}
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
